### PR TITLE
[SPARK-22810][ML][PySpark] Expose Python API for LinearRegression with huber loss.

### PR DESCRIPTION
**Pull request by yanboliang**
## What changes were proposed in this pull request?
Expose Python API for _LinearRegression_ with _huber_ loss.

## How was this patch tested?
Unit test.

### CODE REVIEW

File: python/pyspark/ml/tests.py [hunk 1]
```diff
@@ -1725,6 +1725,27 @@ def test_offset(self):
...
 
 
+class LinearRegressionTest(SparkSessionTestCase):
+
+    def test_linear_regression_with_huber_loss(self):
+
+        data_path = "data/mllib/sample_linear_regression_data.txt"
+        df = self.spark.read.format("libsvm").load(data_path)
+
+        lir = LinearRegression(loss="huber")
```
Comment (WeichenXu123) [inline python/pyspark/ml/tests.py]: The testcase should include `setEpsilon`

File: python/pyspark/ml/regression.py [hunk 2]
```diff
@@ -155,6 +183,14 @@ def intercept(self):
         """
         return self._call_java("intercept")
 
+    @property
+    @since("2.3.0")
+    def scale(self):
+        """
+        The value by which \|y - X'w\| is scaled down when loss is "huber".
```
Comment (WeichenXu123) [inline python/pyspark/ml/regression.py]: add doc "When square loss the value is 1.0"

**Review by WeichenXu123 — APPROVED** [review]

**Comment by WeichenXu123** [comment]
LGTM.

**Comment by yanboliang** [comment]
Merged into master, thanks.

**Pull request closed without merging**